### DISCUSSION TITLE
How to Handle Cors Request

### DISCUSSION THREAD

**Discussion by vimalmistry** [uNetworking/uWebSockets.js#619] [Q&A]
Share your example.

I made an API using Uwebsocket.js. But the front end is simple HTML generated using a static site generator.
I am using Nginx for reverse proxy.

The API call is not working on the browser. it is working in the postman/insomnia and flutter app. but not in the browser.

```nginx

server {
    server_name echo.domain.com;
    location / {
     
        limit_req zone=mylimit;
        server_tokens off;
        proxy_pass http://nodejs;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_http_version 1.1;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $host;

    }



```

```js
 res.writeHeader("Access-Control-Allow-Origin", "*");
   res.writeHeader("Access-Control-Allow-Credentials", "true");
   res.writeHeader("Access-Control-Allow-Methods", "GET, POST, OPTIONS");
   res.writeHeader(
      "Access-Control-Allow-Headers",
      "Origin, X-Requested-With, Content-Type, Accept",
   );
```
![Screenshot 2021-10-02 at 6 01 57 AM](https://user-images.githubusercontent.com/3993163/135698116-54c40cb9-b4b1-4221-87d0-1203c9f756a8.png)

![Screenshot 2021-10-02 at 6 04 42 AM](https://user-images.githubusercontent.com/3993163/135698119-2d490ed5-641a-408b-be8a-7b25047ff07d.png)

I also tried using manual options routes.

![Screenshot 2021-10-02 at 6 05 44 AM](https://user-images.githubusercontent.com/3993163/135698133-5fcf975a-dc73-44e5-bf29-59963bb40976.png)

Please, anyone has a working example. or can guide me.

Or is there a config for the client site? I am using `fetch` and `axios`

**Comment by paradis-A** [comment]
wow your code is super long.
app.get("/*", async (res, req) => {
        res.writeHeader("Access-Control-Allow-Origin", "*") //allow cors
        res.end("hello world")
    })
also works with ws

**Reply by Hiutaky** [reply]
For GET routes works well, not the same for POST, still receiving CORS error

**Reply by Hiutaky** [reply]
Solution for POST non-simple requests. You need to register an options route and allow origin there

https://github.com/uNetworking/uWebSockets.js/discussions/316#discussioncomment-2027652
